patches/git/0001-Initial-Managarm-support.patch

```diff
@@ -0,0 +1,58 @@
+From 53584e9beb2c607d6dd77422db652c6014ddabcb Mon Sep 17 00:00:00 2001
+From: Dennis Bonke <admin@dennisbonke.com>
+Date: Fri, 16 Jul 2021 15:58:28 +0200
+Subject: [PATCH] Initial Managarm support
+
+Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
+---
+ config.mak.uname  | 12 ++++++------
+ git-compat-util.h | 10 ++--------
+ 2 files changed, 8 insertions(+), 14 deletions(-)
+
+diff --git a/config.mak.uname b/config.mak.uname
+index cb443b4e02..6cb9520c54 100644
+--- a/config.mak.uname
++++ b/config.mak.uname
+@@ -4,12 +4,12 @@
+ # Microsoft's Safe Exception Handling in libraries (such as zlib).
+ # Typically required for VS2013+/32-bit compilation on Vista+ versions.
+ 
+-uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
+-uname_M := $(shell sh -c 'uname -m 2>/dev/null || echo not')
+-uname_O := $(shell sh -c 'uname -o 2>/dev/null || echo not')
+-uname_R := $(shell sh -c 'uname -r 2>/dev/null || echo not')
+-uname_P := $(shell sh -c 'uname -p 2>/dev/null || echo not')
+-uname_V := $(shell sh -c 'uname -v 2>/dev/null || echo not')
++uname_S := #$(shell sh -c 'uname -s 2>/dev/null || echo not')
++uname_M := #$(shell sh -c 'uname -m 2>/dev/null || echo not')
++uname_O := #$(shell sh -c 'uname -o 2>/dev/null || echo not')
++uname_R := #$(shell sh -c 'uname -r 2>/dev/null || echo not')
++uname_P := #$(shell sh -c 'uname -p 2>/dev/null || echo not')
++uname_V := #$(shell sh -c 'uname -v 2>/dev/null || echo not')
+ 
+ ifdef MSVC
+ 	# avoid the MingW and Cygwin configuration sections
+diff --git a/git-compat-util.h b/git-compat-util.h
+index a508dbe5a3..efd1fd96f9 100644
+--- a/git-compat-util.h
++++ b/git-compat-util.h
+@@ -1263,14 +1263,8 @@ int open_nofollow(const char *path, int flags);
+ #endif
+ 
+ #ifndef _POSIX_THREAD_SAFE_FUNCTIONS
+-static inline void flockfile(FILE *fh)
+-{
+-	; /* nothing */
+-}
+-static inline void funlockfile(FILE *fh)
+-{
+-	; /* nothing */
+-}
++#define flockfile(fh)
++#define funlockfile(fh)
```

Comment:
Why is this done?

Reply:
With the original code, I got build failures, something about non static declarations if I remember correctly. In an old patch i fixed that like this, but that is so long ago I’m not quite sure why I opted for that approach.

Reply:
The removed code shoud be a no-op already. Try removing this part of the patch and try to resolve errors if they occur.